statsdaemon.go

```diff
@@ -67,17 +66,28 @@ func (a *Percentiles) String() string {
 	return fmt.Sprintf("%v", *a)
 }
 
-var (
-	sanitizeFilter1 = regexp.MustCompile(`\s+`)
-	sanitizeFilter2 = regexp.MustCompile(`\/`)
-	sanitizeFilter3 = regexp.MustCompile(`[^a-zA-Z_\-0-9\.]`)
-)
-
 func sanitizeBucket(bucket string) string {
-	bucket = sanitizeFilter1.ReplaceAllString(bucket, "_")
-	bucket = sanitizeFilter2.ReplaceAllString(bucket, "-")
-	bucket = sanitizeFilter3.ReplaceAllString(bucket, "")
-	return bucket
+	b := make([]byte, len(bucket))
+	var bl int
+
+	for i := 0; i < len(bucket); i++ {
+		c := bucket[i]
+		switch {
+		// a-z or A-Z or 0-9  or - or . or _
+		case (c >= 97 && c <= 122) || (c >= 65 && c <= 90) || (c >= 48 && c <= 57) || c == 45 || c == 46 || c == 95:
```

Comment:
I think you might be able to do something like 

```go
c >= byte('a')
```

instead of 

```go
c >= 97
```
for dat readability